Libraries/Core/InitializeCore.js

```diff
@@ -116,6 +116,23 @@ if (!global.__fbDisableExceptionsManager) {
   ErrorUtils.setGlobalHandler(handleError);
 }
 
+const formatVersion = version =>
+  `${version.major}.${version.minor}.${version.patch}` +
+  (version.prerelease !== null ? `-${version.prerelease}` : '');
+
+const ReactNativeVersion = require('./ReactNativeVersion');
+const nativeVersion = require('NativeModules').PlatformConstants.reactNativeVersion;
+if (ReactNativeVersion.version.major !== nativeVersion.major ||
```

Comment:
should be keep this behind the __DEV__ flag?

Reply:
@ide prefered not to, I don't really have a preference

Reply:
The main thing is to keep dev and prod similar, particularly around edge cases, error handling, initialization, cleanup. If we move this to a `__DEV__`-only check, we should change this to a console.error call instead of throwing an error.

Reply:
i would rather let it continue in prod than spectacularly failing at startup. idk